crates/uv-client/src/base_client.rs

```diff
@@ -88,7 +94,11 @@ impl<'a> BaseClientBuilder<'a> {
         self
     }
 
-    pub fn build(self) -> BaseClient {
+    pub fn is_offline(&self) -> bool {
+        matches!(self.connectivity, Connectivity::Offline)
+    }
+
+    pub fn build(&self) -> BaseClient {
```

Comment:
In lieu of calling `build` multiple times, did you explore trying to do this with interior mutability? E.g., making `client()` a `OnceCell` or something lazily constructed?

Reply:
Ah I looked at this a little in #2439 — I'd be happy to explore that more next.